### PR TITLE
Add nameID argument to wpsimplesaml_match_user

### PR DESCRIPTION
**Pull request by Gazi96**

### CODE REVIEW

**Review by tomjn — APPROVED** [review]

File: inc/namespace.php [hunk 1]
```diff
@@ -433,7 +434,7 @@ function get_or_create_wp_user( \OneLogin\Saml2\Auth $saml ) {
 	 *
```
Comment (mikelittle) [inline inc/namespace.php:434]: @Gazi96 Rafal, 
could you update the documentation block for the `wpsimplesaml_match_user` filter, to document the extra parameters, please?

**Review by mikelittle — CHANGES_REQUESTED** [review]

File: inc/namespace.php (hunk 1 above)
Comment (Gazi96) [reply inc/namespace.php:435]: @mikelittle, good point, I have updated documentation.

File: inc/namespace.php [hunk 2]
```diff
@@ -430,10 +431,11 @@ function get_or_create_wp_user( \OneLogin\Saml2\Auth $saml ) {
...
 	 * @param string $email      Email from SAMLResponse
 	 * @param array  $attributes SAML Attributes parsed from SAMLResponse
+	 * @param string $name_id    Name ID from SAMLResponse
 	 *
 	 * @return null|false|\WP_User User object or false if not found
 	 */
-	$user = apply_filters( 'wpsimplesaml_match_user', null, $email, $attributes );
+	$user = apply_filters( 'wpsimplesaml_match_user', null, $email, $attributes, $name_id );
```
Comment (devmons) [inline inc/namespace.php:438]: Good catch 👏

**Review by devmons — APPROVED** [review]

**Review by mikelittle — APPROVED** [review]
Looks good. Thanks for this.

**Pull request merged**